packages/lesmis/src/api/queries/useDashboardReportData.js

```diff
@@ -43,16 +46,31 @@ const getReportCodesByCode = dashboards => {
   }, {});
 };
 
+const getFavouriteStatus = (userFavouriteDashboardItems, userId, reportCode) => {
+  if (!userId) {
+    return IDLE;
+  }
+
+  const isFavouriteDashboardItemFound = userFavouriteDashboardItems.find(
+    dashboardItemCode => dashboardItemCode === reportCode,
+  );
+
+  return isFavouriteDashboardItemFound ? IS_FAVOURITE : IS_NOT_FAVOURITE;
+};
+
 export const useDashboardReportDataWithConfig = ({
   entityCode,
   reportCode,
   startDate,
   endDate,
 }) => {
+  const { userId } = useUser();
+
   const query = combineQueries({
     reportData: useDashboardReportData({ entityCode, reportCode, startDate, endDate }),
     dashboards: useDashboardData({ entityCode }),
   });
+  const favouriteDashboardItems = useFavouriteDashboardItem({ userId })?.data;
```

Comment:
Feels a little awkward that it's the frontend's responsibility to map which dashboards the user has favourited, ideally this is managed by the server.

However, since we just query through to central-server via lesmis-server, I feel this is the cleanest way for now 👍 

One day I'd love to switch the central-server to graphql, then we could do all the data joins we need in a single query!